corehq/pillows/management/commands/utils.py

```diff
@@ -5,16 +5,44 @@
 
 
 def pprint(obj, namespace={}, stream=sys.stdout, indent=4, separators=(',', ': ')):
+    """Pretty-print an Elastic mapping. Like `json.dump()`, but output valid
+    Python code rather than JSON.
+
+    :param obj: object to be 'printed'
+    :param namespace: (optional dict) see ElasticMappingEncoder `namespace` arg
+    :param stream: (optional) file-like object where output is written,
+                   default=`sys.stdout`
+    :param indent: (int) see `json.dump`
+    :param separators: (tuple) see `json.dump`
+    """
     encoder = ElasticMappingEncoder(namespace, indent=indent, separators=separators)
     for chunk in encoder.iterencode(obj):
         stream.write(chunk)
 
 
 class ElasticMappingEncoder(json.JSONEncoder):
+    """Encodes an Elastic mapping (nested python dict) as properly indented
+    python code.
 
+    A JSONEncoder which writes 'True', 'False' and 'None' instead of 'true',
```

Comment:
This is probably a naive question, but the description here sounds a lot like just calling `pprint.pp` (or one of the other pretty print functions) on the dict object, instead of `json.dump`. What are differences between this implementation and that?

Reply:
I investigated both the `pprint` module, as well as `IPython.lib.pretty` (used by Django shell) prior to this implementation. Neither of those modules produce this familiar, readable format. Also, neither are extensible (would have required re-writing their object traversal logic to change the whitespace, which defeats the purpose).